__init__.py

```diff
@@ -197,6 +197,28 @@ def _handle_switch(self, message):
         self.log.debug("Entity: %s" % entity)
         self.log.debug("Action: %s" % action)
 
+        # Handle turn on/off all intent
+        try:
+            if self.voc_match(entity,"all_lights"):
+                domain = "light"
+            elif self.voc_match(entity,"all_switches"):
+                domain = "switch"
+            else:
+                domain = None
+
+            if not domain is None:
+                ha_entity = {'dev_name': entity}
+                ha_data = {'entity_id': 'all'}
+
+                self.ha.execute_service(domain, "turn_%s" % action, ha_data)
+                self.speak_dialog('homeassistant.device.%s' % action, data=ha_entity)
+                return
+        # TODO: need to figure out, if this indeed throws a KeyError
```

Comment:
if we're not sure, or it is too difficult to isolate, we could catch for generic `Exception`?

Reply:
OK